test/functional/api/extmark_spec.lua

```diff
@@ -1599,12 +1599,71 @@ describe('API/extmarks', function()
     set_extmark(ns, 3, 0, 0, { sign_text = '>>' })
     set_extmark(ns, 4, 0, 0, { virt_text = {{'text', 'Normal'}}})
     set_extmark(ns, 5, 0, 0, { virt_lines = {{{ 'line', 'Normal' }}}})
-    eq(5, #get_extmarks(-1, 0, -1, { details = true }))
+    eq(5, #get_extmarks(-1, 0, -1, {}))
     eq({{ 2, 0, 0 }}, get_extmarks(-1, 0, -1, { type = 'highlight' }))
     eq({{ 3, 0, 0 }}, get_extmarks(-1, 0, -1, { type = 'sign' }))
     eq({{ 4, 0, 0 }}, get_extmarks(-1, 0, -1, { type = 'virt_text' }))
     eq({{ 5, 0, 0 }}, get_extmarks(-1, 0, -1, { type = 'virt_lines' }))
   end)
+
+  it("invalidated marks are deleted", function()
+    screen = Screen.new(40, 6)
+    screen:attach()
+    feed('dd6iaaa bbb ccc<CR><ESC>gg')
+    set_extmark(ns, 1, 0, 0, { sign_text = 'S1' })
+    set_extmark(ns, 2, 1, 0, { invalidate = true, sign_text = 'S2' })
+    -- mark with invalidate is removed
+    command('2d')
+    screen:expect([[
+      S1aaa bbb ccc                           |
+        ^aaa bbb ccc                           |
+        aaa bbb ccc                           |
+        aaa bbb ccc                           |
+        aaa bbb ccc                           |
+                                              |
+    ]])
+    -- mark is restored with undo_restore == true
+    command('silent undo')
+    screen:expect([[
+      S1aaa bbb ccc                           |
+      S2^aaa bbb ccc                           |
+        aaa bbb ccc                           |
+        aaa bbb ccc                           |
+        aaa bbb ccc                           |
+                                              |
+    ]])
+    -- mark is deleted with undo_restore == false
+    set_extmark(ns, 1, 0, 0, { invalidate = true, undo_restore = false, sign_text = 'S1' })
+    set_extmark(ns, 2, 1, 0, { invalidate = true, undo_restore = false, sign_text = 'S2' })
+    command('1d 2')
+    eq(0, #get_extmarks(-1, 0, -1, {}))
+    set_extmark(ns, 0, 0, 4, { invalidate = true, undo_restore = false,
+                               hl_group = 'Error', end_col = 7 })
+    eq({{3, 0, 4}}, get_extmarks(-1, 0, -1, {}))
+    -- mark is not removed when deleting bytes before the range
+    feed('dw')
+    eq(3, get_extmarks(-1, 0, -1, { details = true })[1][4].end_col)
```

Comment:
could use `get_buf_get_extmark_id`. Doesn't really matter technically if there is just one extmark, but signals the intent we are verifying the state of one specific extmark. imo.